From d66e24e149e4630e72a58b253c7f68f67664e952 Mon Sep 17 00:00:00 2001
From: Robert Beckett <bob.beckett@collabora.com>
Date: Fri, 7 Mar 2025 18:24:59 +0000
Subject: [PATCH 025/188] drm/ttm: allow min calculated page pool size

For a given device's use case, there can be a minimum required
size of gtt required. Allow the user to set a minimum so that
the same setup can be used across a range of products without
having to set gtt.size specifically for each based on RAM.

Signed-off-by: Robert Beckett <bob.beckett@collabora.com>
---
 drivers/gpu/drm/ttm/ttm_device.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/gpu/drm/ttm/ttm_device.c b/drivers/gpu/drm/ttm/ttm_device.c
index d3bfb9a69..7aecd8968 100644
--- a/drivers/gpu/drm/ttm/ttm_device.c
+++ b/drivers/gpu/drm/ttm/ttm_device.c
@@ -40,6 +40,9 @@
 #include "ttm_module.h"
 #include "ttm_bo_internal.h"
 
+static unsigned long ttm_pages_min;
+MODULE_PARM_DESC(pages_min, "Minimum calculated limit for the allocated pages");
+module_param_named(pages_min, ttm_pages_min, ulong, 0644);
 /*
  * ttm_global_mutex - protecting the global state
  */
@@ -91,6 +94,9 @@ static int ttm_global_init(void)
 	num_pages = ((u64)si.totalram * si.mem_unit) >> PAGE_SHIFT;
 	num_pages /= 2;
 
+	/* Ensure it meats minimum */
+	num_pages = max(num_pages, ttm_pages_min);
+
 	/* But for DMA32 we limit ourself to only use 2GiB maximum. */
 	num_dma32 = (u64)(si.totalram - si.totalhigh) * si.mem_unit
 		>> PAGE_SHIFT;
-- 
2.55.0

